de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/fskdbview/FSKDBViewNodeModel.java

```diff
@@ -185,9 +185,41 @@ public void saveCurrentValue(NodeSettingsWO content) {
   @Override
   protected PortObject[] performExecute(PortObject[] inObjects, ExecutionContext exec)
       throws Exception {
+    PortObject outport = inObjects[0];
     synchronized (getLock()) {
+      FSKDBViewRepresentation representation = getViewRepresentation();
+      if (outport == null) {
+        // if the optional input port is not provided then
+        outport = createEmptyTable(exec);
+      } else if (representation.getTable() == null) {
+
+        // construct a BufferedDataTable from the input object.
+        BufferedDataTable table = (BufferedDataTable) inObjects[0];
+        JSONDataTable jsonTable = JSONDataTable.newBuilder().setDataTable(table).build(exec);
+        representation.setTable(jsonTable);
+
+        // set the table ID which will be used in broadcasting and receiving event in the component.
+        String connectedNodeId = getTableId(0);
+        representation.setTableID(connectedNodeId);
+
+        outport = inObjects[0];
+      }
     }
-    return new PortObject[] {inObjects[0]};
+    return new PortObject[] {outport};
+  }
+
+  /**
+   * A helper method for creating an empty table in the case of empty input port.
+   * 
+   * @param ExecutionContext exec.
+   * @return an empty BufferedDataTable instance.
+   */
+  private BufferedDataTable createEmptyTable(final ExecutionContext exec) {
+    BufferedDataContainer container =
+        exec.createDataContainer(new DataTableSpecCreator().createSpec());
+    container.close();
+    BufferedDataTable out = container.getTable();
```

Comment:
We should not declare local variables and return them immediately after. We should return instead *container.getTable()*. https://rules.sonarsource.com/java/RSPEC-1488?search=local%20variables%20should%20not%20be%20declared%20and%20then%20immediately%20returned